lib/puppet/util.rb

```diff
@@ -758,6 +758,13 @@ def deterministic_rand_int(seed,max)
     Random.new(seed).rand(max)
   end
   module_function :deterministic_rand_int
+
+  def skip_external_facts
```

Comment:
this should check first if `load_external` is defined on Facter
`Facter.respond_to?(:load_external)`